vital_sqi/data/signal_sqi_class.py

```diff
@@ -13,7 +19,116 @@ def __init__(self, wave_type=None, signals=None, sampling_rate=None,
         self.wave_type = wave_type
         self.sqi_indexes = sqi_indexes
         self.info = info
+        self.isSplit = False
+        self.isFiltered = False
+        self.SQIComputed = False
+        self.unfilteredsignal = None
+
+    def split_to_segments(self, minute_remove=0.5, segment_duration=30):
+        """
+
+        Parameters
+        ----------
+        minute_remove : float
+        number of minutes to be removed from start and end of the signal
+
+        segment_duration : int
+        duration of segment in seconds            
+
+        Returns
+        -------
+        object of SignalSQI class
+        
+        """
+        if self.signals.ndim == 1 and not self.isSplit:
+            #Compute size of to be removed slice and each segment
+            removal = round(minute_remove*60*self.sampling_rate)
+            segment_size = segment_duration*self.sampling_rate
+            #Remove the excess parts at the start and end according to calculate size of removal
+            tmp_signal = self.signals[removal:-removal]
+            #Calculate how many segments can fit into remaining data and reshape the array
+            excess = len(tmp_signal) % segment_size
+            tmp_signal = np.reshape(tmp_signal[:-excess], (segment_size, -1), 'F').T
+            self.isSplit = True
+            self.unfilteredsignal = tmp_signal
+            self.update_signal(tmp_signal)
+        elif not self.isSplit:
+            raise Exception("Not supported for multidimensional signals")
+        else:
+            return self
+
+
+    def filter_preprocess(self, hp_cutoff_order=(1, 1), lp_cutoff_order=(20, 4), filter_type='butter'):
```

Comment:
I think it is better to leave this function in its original place and we call it from common.